plugins/out_s3/s3_store.c

```diff
@@ -175,6 +176,14 @@ int s3_store_buffer_put(struct flb_s3 *ctx, struct s3_file *s3_file,
             return -1;
         }
         s3_file->fsf = fsf;
+        if (file_first_log_time == NULL) {
+            s3_file->first_log_time = time(NULL);
+            flb_plg_error(ctx->ins, "Failed to get timestamp used for the S3 key, "
+                          "use the current time instead.");
+        } else {
+            s3_file->first_log_time = file_first_log_time;
+        }
+
```

Comment:
Will file_first_log_time ever be null here? Please find an example of how that can happen.

Can you resolve the example by moving this code to the place where you evaluate the first_log_time?